### PR TITLE
Add Docs and Examples and helper methods to `PhysicalSortExpr` 

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?
Part of https://github.com/apache/datafusion/issues/12446

Closes #.

## Rationale for this change

Basically while working on https://github.com/apache/datafusion/pull/12562 I got annoyed at the lack of documentation and display implementations as well as how award it was to create `PhysicalSortExprs`.

## What changes are included in this PR?
1. Added a bunch of docs and display impls and examples

## Are these changes tested?
Yes, by CI and doc tests

## Are there any user-facing changes?
Better docs, easier to use APIs.

No API changes

### CODE REVIEW

File: datafusion/core/src/datasource/listing/table.rs [hunk 1]
```diff
@@ -1238,20 +1238,13 @@ mod tests {
...
-                        },
-                    },
-                    PhysicalSortExpr {
-                        expr: physical_col("int_col", &schema).unwrap(),
-                        options: SortOptions {
-                            descending: true,
-                            nulls_first: true,
-                        },
-                    },
+                    PhysicalSortExpr::new_default(physical_col("string_col", &schema).unwrap())
```
Comment (alamb) [inline datafusion/core/src/datasource/listing/table.rs:1241]: This shows what I would like to be able to do

File: datafusion/physical-expr-common/src/sort_expr.rs [hunk 2]
```diff
@@ -30,6 +30,51 @@ use datafusion_common::Result;
 use datafusion_expr_common::columnar_value::ColumnarValue;
 
 /// Represents Sort operation for a column in a RecordBatch
+///
+/// Example:
```
Comment (alamb) [inline datafusion/physical-expr-common/src/sort_expr.rs:34]: This looks like a massive example but it renders neatly I think:

Screenshot 2024-09-23 at 8 56 58 AM

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb

**Comment by alamb** [comment]
Thanks for the quick review @comphead

**Pull request merged**